libs/sdk-core/src/models.rs

```diff
@@ -661,14 +663,29 @@ impl From<OpeningFeeParams> for grpc::OpeningFeeParams {
         Self {
             min_msat: ofp.min_msat,
             proportional: ofp.proportional,
-            valid_until: ofp.valid_until,
+            valid_until: ofp.valid_until.to_string(),
             max_idle_time: ofp.max_idle_time,
             max_client_to_self_delay: ofp.max_client_to_self_delay,
             promise: ofp.promise,
         }
     }
 }
 
+impl TryFrom<grpc::OpeningFeeParams> for OpeningFeeParams {
+    type Error = anyhow::Error;
+
+    fn try_from(gofp: grpc::OpeningFeeParams) -> std::result::Result<Self, Self::Error> {
+        Ok(Self {
+            min_msat: gofp.min_msat,
+            proportional: gofp.proportional,
+            valid_until: serde_json::from_str(&gofp.valid_until).map_err(|e| anyhow!(e))?,
```

Comment:
This error will propagate all the way up and prevent from this lsp to be returned to the user.
Let's add a meaningful error here such as "lsp <id> `valid_until` field format is violates the specification"

Reply:
I am tending towards moving the fee params validation code to this point. I have noticed we are validating on `get_longest_fee_params` and get `cheapest_fee_params` methods and moving the validation to this point is earlier and holds for all use cases. We can omit LSP that doesn't pass the validation.
@ok300 @JssDWt what do you think?

Reply:
Yeah, it's a message spec, so the place to validate whether the message has a valid shape is here.